src/annotator/plugin/toolbar.coffee

```diff
@@ -26,6 +27,15 @@ module.exports = class Toolbar extends Plugin
       $(@options.container).append @toolbar
     else
       $(@element).append @toolbar
+    
+    # https://h.readthedocs.io/projects/client/en/latest/publishers/config/?highlight=onLoginRequest#cmdoption-arg-showhighlights
+    highlightsAreVisible = true; # default to on
+    showHighlightsConfig = window.hypothesisConfig().showHighlights;
```

Comment:
`window.hypothesisConfig()` will exist only if the page supplies a custom Hypothesis client configuration and chooses to use this mechanism. Most pages don't supply such a configuration and many will use a JSON script tag. See https://h-client.readthedocs.io/en/latest/publishers/config/#configuring-the-client-using-json

Aside from this issue, I see a broader problem with the approach which is that this introduces multiple places in the application that are trying to interpret the client configuration and map it to the highlight visibility state. If some additional logic is introduced in future which changes how that works in one place, it could get out of sync with the other.

From reading the code in this file and `guest.coffee`, it looks like the `Guest` instance's `visibleHighlights` property (`@annotator.visibleHighlights` in this file, `@visibleHighlights` in `guest.coffee`) is what the current code treats as the source of truth.